src/elements/topographic-profile.ts

```diff
@@ -0,0 +1,138 @@
+import {LitElementI18n} from '../i18n.js';
+import {html} from 'lit';
+import {customElement, state} from 'lit/decorators.js';
+import i18next from 'i18next';
+import draggable from './draggable';
+import ToolboxStore from '../store/toolbox';
+import {cartesianToLv95} from '../projection';
+import * as d3 from 'd3';
+
+import type {GeometryAction} from '../store/toolbox';
+
+type ProfileServiceFormat = 'json' | 'csv'
+
+@customElement('topographic-profile')
+export class TopographicProfile extends LitElementI18n {
+  @state() hidden = true;
+  private coords: number[][] | undefined;
+  private name: string | undefined;
+
+  constructor() {
+    super();
+    super.hidden = this.hidden;
+    ToolboxStore.geometryAction.subscribe(options => this.handleActions(options));
+  }
+
+  connectedCallback() {
+    draggable(this, {
+      allowFrom: '.drag-handle'
+    });
+    super.connectedCallback();
+  }
+
+  update(changedProperties) {
+    if (changedProperties.has('hidden')) {
+      super.hidden = this.hidden;
+    }
+    super.update(changedProperties);
+  }
+
+  async handleActions(options: GeometryAction) {
+    if (options.action === 'profile') {
+      const geom = ToolboxStore.geometries.getValue().find(geom => geom.id === options.id);
+      this.coords = geom!.positions.map(c => cartesianToLv95(c));
+      this.name = geom!.name;
+      this.hidden = false;
+      await this.updateComplete;
```

Comment:
Why do we need to await?

Reply:
We need to wait for the element to render/update in order for d3 to select the element to draw the graph